src/snowflake/snowpark/mock/_plan.py

```diff
@@ -1376,6 +1386,97 @@ def outer_join(base_df):
             res.append(len(deleted_row_idx))
 
         return [Row(*res)]
+    elif isinstance(source_plan, Pivot):
+        child_rf = execute_mock_plan(source_plan.child)
+
+        assert (
+            len(source_plan.aggregates) == 1
+        ), "Dataframe plan should fail before this if one aggregate isn't supplied."
+        agg = source_plan.aggregates[0]
+        assert (
+            len(agg.children) == 1
+        ), "Aggregate functions should take exactly one parameter."
+        agg_column = plan.session._analyzer.analyze(agg.children[0])
+
+        agg_function_name = agg.name.lower()
+        agg_functions = {
+            "avg": statistics.mean,
+            "count": len,
+            "max": max,
+            "min": min,
+            "sum": sum,
+        }
+
+        if agg_function_name not in agg_functions:
+            raise ValueError(
+                f"Unsupported pivot aggregation function {agg_function_name}."
+            )
+
+        pivot_column = plan.session._analyzer.analyze(source_plan.pivot_column)
+
+        if isinstance(source_plan.pivot_values, Iterable):
+            pivot_values = [exp.value for exp in source_plan.pivot_values]
+        elif source_plan.pivot_values is None:
+            pivot_values = []
+        else:
+            analyzer.session._conn.log_not_supported_error(
+                external_feature_name=f"Pivot values from {source_plan.pivot_values}",
+                internal_feature_name=type(source_plan).__name__,
+                raise_error=NotImplementedError,
+            )
+
+        grouping_columns = [
+            plan.session._analyzer.analyze(c) for c in source_plan.grouping_columns
+        ]
+        indices = grouping_columns or [
+            col for col in child_rf.keys() if col not in {agg_column, pivot_column}
+        ]
```

Comment:
just for my own learning purpose, grouping_columns are columns specified in `order by` and if `order by` not specified then it will use the rest of the columns for grouping (excluding agg and pivot column) ?

Reply:
Correct. That behavior is not documented on the main pivot page, but it is documented [here](https://community.snowflake.com/s/article/Pivot-returns-more-rows-than-expected).

Reply:
nit: can you help document this behavior in the code as comment? so that in the future when someone needs to work on this they can quickly ramp up